auth/template.yaml

```diff
@@ -2010,16 +2008,123 @@ Resources:
         - Key: System
           Value: Authentication
 
+  SharedSignalsHealthCheckFunction:
+    Condition: IsNotProduction
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub ${AWS::StackName}-shared-signals-health-check
+      Description: Shared Signals health check function
+      CodeUri: shared-signals-health-check/
+      Handler: app.lambdaHandler
+      Role: !GetAtt SharedSignalsHealthCheckFunctionIAMRole.Arn
+      KmsKeyArn: !GetAtt SharedSignalsHealthCheckFunctionKMSKey.Arn
+      Events:
+        EventBridgeEvent:
+          Type: ScheduleV2
+          Properties:
+            ScheduleExpression: rate(10 minutes)
+            RetryPolicy:
+              MaximumRetryAttempts: 2
+              MaximumEventAgeInSeconds: 60
+      Environment:
+        Variables:
+          REGION: !Ref AWS::Region
+          HEALTH_CHECK_TOKEN_URL: !Sub '{{resolve:ssm:/${ConfigStackName}/shared-signal/health-check-token-url}}'
+          HEALTH_CHECK_VERIFY_URL: !Sub '{{resolve:ssm:/${ConfigStackName}/shared-signal/health-check-verify-url}}'
+          HEALTH_CHECK_SECRET_NAME: !Sub /${ConfigStackName}/shared-signal/health-check-secrets
+      Tags:
+        Product: GOV.UK
+        Environment: !Ref Environment
+        System: Authentication
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: es2020
+        SourceMap: false
+        EntryPoints:
+          - app.ts
+
+  SharedSignalsHealthCheckFunctionKMSKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: KMS key for encrypting shared signals health check function secrets
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action:
+              - kms:*
+            Resource:
+              - '*'
+          - Effect: Allow
+            Principal:
+              Service: !Sub lambda.amazonaws.com
+            Action:
+              - kms:Encrypt*
+              - kms:Decrypt*
+            Resource:
+              - '*'
```

Comment:
Should this reference the lambda function? I appreciate we may not do this elsewhere

Reply:
We don't, but I will check

Reply:
Sorted ✔️